glean-core/android/src/main/java/mozilla/telemetry/glean/config/Configuration.kt

```diff
@@ -17,12 +17,13 @@ import mozilla.telemetry.glean.net.PingUploader
  * **CAUTION**: This must match _exactly_ the definition on the Rust side.
  *  If this side is changed, the Rust side need to be changed, too.
  */
-@Structure.FieldOrder("dataDir", "packageName", "uploadEnabled", "maxEvents")
+@Structure.FieldOrder("dataDir", "packageName", "uploadEnabled", "maxEvents", "deferCollection")
 internal class FfiConfiguration(
     dataDir: String,
     packageName: String,
     uploadEnabled: Boolean,
-    maxEvents: Int? = null
+    maxEvents: Int? = null,
+    deferCollection: Boolean
```

Comment:
I think `deferCollection` is a bit unclear: we're not really deferring the collection, we're deferring writing to disk, but only for `lifetime: metrics`.

At the very least, this should be `deferPersistence`. Maybe even`delayPingLifetimeIO`? @badboy ?

Reply:
I am happy with `delayPingLifetimeIO`, should I just change it to this and we can discuss this further on my next PR? Specially since this prop might change to an interval for the next one?

Reply:
Yes, let's change it here. Depending on how we go for it, we might not really need to offer ways to customise the timeout ;) But let's talk about that in the next PR :)